uni/lib/model/providers/library_reservations_provider.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:async';
+
+import 'package:logger/logger.dart';
+import 'package:uni/controller/fetchers/library_reservation_fetcher.dart';
+import 'package:uni/controller/local_storage/app_library_reservation.dart';
+import 'package:uni/model/entities/library_reservation.dart';
+import 'package:uni/model/entities/session.dart';
+import 'package:uni/model/providers/state_provider_notifier.dart';
+import 'package:uni/model/request_status.dart';
+
+class LibraryReservationsProvider extends StateProviderNotifier {
+  List<LibraryReservation>? _reservations;
```

Comment:
_reservations shouldn't be nullable at all. If a request or the database returns nothing, it should be an empty list.